spec/unit/action/deploy/environment_spec.rb

```diff
@@ -265,6 +265,19 @@
       describe "deployment purge level" do
         let(:purge_levels) { [:deployment] }
 
+        describe "to make r10k deploy pools thread safer" do
```

Comment:
I'm honestly not even sure this test is worth it. If literally anything changes about what we do here, we'll have to update this. I'm fine leaving it, but I'm considering the testing for this addition to be really accomplished at a much higher leve.

Reply:
I think I agree. This test doesn't really seem to provide any value and we'll need more robust testing of this anyway.

Reply:
I'm not personally a huge fan of this test as well, but I think it provides a sanity check to make sure we call `reload!` before purging.